File: src/core/Elsa.Core/Metadata/ActivityPropertyDescriptor.cs
namespace Elsa.Metadata
{
    public class ActivityPropertyDescriptor
    {
        public ActivityPropertyDescriptor(string name, string type, string label, string hint = null, object options = null)
        {
            Name = name;
            Type = type;
            Label = label;
            Hint = hint;
            Options = options;
        }
        
        public string Name { get; }
        public string Type { get; }
        public string Label { get; }
        public string Hint { get; }
        public object Options { get; }
    }
}

File: src/designer/elsa-workflows-studio/src/modelsCopy/activity-property-descriptor.ts
export interface ActivityPropertyDescriptor {
  name: string;
  type: string;
  label: string;
  hint?: string;
  options?: any;

  uiHint?: string;
  category?: string;
  defaultValue?: any;
  defaultSyntax?: string;
  supportedSyntaxes?: Array<string>;
  isReadOnly?: boolean;
  defaultWorkflowStorageProvider?: string;
  disableWorkflowProviderSelection?: boolean;
  considerValuesAsOutcomes?: boolean;
}
